### ISSUE TITLE
Trial 2 failed, because the value None could not be cast to float.

### ISSUE THREAD

**Issue by manza-ari** [optuna/optuna#3715]
### Expected behavior

I want to find optimal milestones (SGD optimizer) for my model but facing above error 

`
# OPTUNA Learning Rate and Optimizer 

                optuna_milestone_min =  optuna_trial.suggest_int('milestones' , 1, 30, step = 5)
                optuna_milestone_max =  optuna_trial.suggest_int('milestones' , 1, 30, step = 5 )
                
                
                # Loss, criterion and scheduler (re)initialization
                criterion      = nn.CrossEntropyLoss(reduction='none')
                optim_backbone = optim.SGD(models.parameters(), lr=LR, 
                    momentum=MOMENTUM, weight_decay=WDECAY)                             #OPTUNA Learning Rate suggestions 
    
                sched_backbone = lr_scheduler.MultiStepLR(optim_backbone, milestones=[optuna_milestone_min, optuna_milestone_max])
                optimizers =  optim_backbone
                schedulers = sched_backbone
`

### Environment

- Optuna version:  2.10.0
- Python version: 3.8
- OS: Ubuntu 
- (Optional) Other libraries and their versions:

### Error messages, stack traces, or logs

```shell
Trial 2 failed, because the value None could not be cast to float.
```

### Steps to reproduce

1.
2.
3.
```pmodels      = resnet18
                
                torch.backends.cudnn.benchmark = True
                models = torch.nn.DataParallel(models, device_ids=[0])
                
                # OPTUNA Learning Rate and Optimizer 

                optuna_milestone_min =  optuna_trial.suggest_int('milestones' , 1, 30, step = 5)
                optuna_milestone_max =  optuna_trial.suggest_int('milestones' , 1, 30, step = 5 )
                
                
                # Loss, criterion and scheduler (re)initialization
                criterion      = nn.CrossEntropyLoss(reduction='none')
                optim_backbone = optim.SGD(models.parameters(), lr=LR, 
                    momentum=MOMENTUM, weight_decay=WDECAY)                             #OPTUNA Learning Rate suggestions 
    
                sched_backbone = lr_scheduler.MultiStepLR(optim_backbone, milestones=[optuna_milestone_min, optuna_milestone_max])
                optimizers =  optim_backbone
                schedulers = sched_backbone
                
```

**Comment by nzw0301** [comment]
I'm sure that this is not the bug of Optuna; the objective function returns `None` instead of `float` as in the message. 

Since the code is reproducible and the objective function is not in the issue description, we could not investigate this issue. However could you take a look at the following comments:

1. Use unique param name for min/max milestone. `optuna_milestone_max =  optuna_trial.suggest_int('milestones' , 1, 30, step = 5 )` should be `optuna_milestone_max =  optuna_trial.suggest_int('milestones_max' , 1, 30, step = 5 )`
2. As in the docs of [`MultiStepLR`](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.MultiStepLR.html), `milestones` should increase. In the current implementation, `milestones` might decrease.

**Comment by manza-ari** [comment]
Thank you so much for your response!
Yeah, right. I agree with you, I did the following changes so OPTUNA can suggest an optimal value for the start and end of milestones for the optimizer.  but still facing the same error. 
`
optuna_milestone_min =  optuna_trial.suggest_int('milestones_min' , 10, 15, step = 5)
optuna_milestone_max =  optuna_trial.suggest_int('milestones_max' , 15, 30, step = 5 )
`
I want your comment on one more thing. I want to ask OPTUNA for an optimal initial learning rate as well as the reduction during the given milestone[initial, end] as given here https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.MultiStepLR.html     for some 300 Epochs

`
# OPTUNA Learning Rate and Optimizer 

optuna_initial_lr = optuna_trial.suggest_float('lr', 1e-3, 1e-1, log=True)
optuna_reductions = optuna_trial.suggest_int('num_reduction', 1, 4)
         for i in range(optuna_reductions):
              i = optuna_trial.suggest_int("when_{}".format(i), 50*(i-1), 300, step=50)       
`

**Comment by okra01** [comment]
I'm facing the same problem now. May I ask did you solved it?
